### PR TITLE
serhii.trokhymets

### PR DESCRIPTION
**Pull request by SerhiiTrokhymets**

### CODE REVIEW

File: serhii.trokhymets/practica_lesson/image.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+require 'faraday'
```
Comment (Flippylolz) [inline serhii.trokhymets/practica_lesson/image.rb:3]: Please fix rubocop issues so I can review th code

**Review by Flippylolz — CHANGES_REQUESTED** [review]

File: serhii.trokhymets/practica_lesson/image.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+require 'faraday'
+require 'open-uri'
+require 'uri'
+
+class Image
+  def download(url)
+    result = Faraday.get (url)
```
Comment (Flippylolz) [inline serhii.trokhymets/practica_lesson/image.rb]: Please fix this rubocop error

File: serhii.trokhymets/practica_lesson/image.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    # raise ArgumentError, "code #{result.status}" if url =~URI::regexp
+    # raise ArgumentError.new('not an image') if response is not an image
+    # raise ArgumentError, 'Argument is not numeric' unless url =~URI::regexp
+    raise ArgumentError, "code #{result.status}" if result.status != 200
+
+    # url =~ URI::regexp
+    File.open('image.png', 'wb') { |fp| fp.write(result.body) }
```
Comment (Flippylolz) [inline serhii.trokhymets/practica_lesson/image.rb]: file extension needs to be dynamic

File: serhii.trokhymets/practica_lesson/image.rb [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+require 'uri'
+
+class Image
+  def download(url)
+    result = Faraday.get (url)
+    # raise ArgumentError, "code #{result.status}" if url =~URI::regexp
+    # raise ArgumentError.new('not an image') if response is not an image
+    # raise ArgumentError, 'Argument is not numeric' unless url =~URI::regexp
+    raise ArgumentError, "code #{result.status}" if result.status != 200
+
```
Comment (Flippylolz) [inline serhii.trokhymets/practica_lesson/image.rb:12]: You need to raise TypeError as well

**Review by Flippylolz — CHANGES_REQUESTED** [review]

**Review by Flippylolz — APPROVED** [review]

**Review by slenderock — APPROVED** [review]

**Pull request merged**